cube/cube.c

```diff
@@ -292,6 +292,19 @@ void dumpVec4(const char *note, vec4 vector) {
     fflush(stdout);
 }
 
+char const* to_string(VkPhysicalDeviceType const type)
+{
+    switch(type)
+    {
+        case VK_PHYSICAL_DEVICE_TYPE_OTHER: return "Other";
+        case VK_PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU: return "IntegratedGpu";
+        case VK_PHYSICAL_DEVICE_TYPE_DISCRETE_GPU: return "DiscreteGpu";
+        case VK_PHYSICAL_DEVICE_TYPE_VIRTUAL_GPU: return "VirtualGpu";
+        case VK_PHYSICAL_DEVICE_TYPE_CPU: return "Cpu";
+        default: return "Invalid";
```

Comment:
Probably better to say 'Unknown' than invalid, since it could be that there is a new type added and this code wasn't updated.